api/src/main/java/bio/terra/tanagra/service/search/SqlVisitor.java

```diff
@@ -63,22 +63,32 @@ public String createSql(Query query) {
             .put("table", underlayResolver.resolveTable(query.primaryEntity()))
             .put("filter", filterSql.orElse("TRUE"))
             .build();
-    String sqlWithoutOrderBy = StringSubstitutor.replace(template, params);
 
+    String sqlQuery = StringSubstitutor.replace(template, params);
     if (query.orderBy() != null) {
       String orderByExpression = query.orderBy().accept(new SelectionVisitor(searchContext));
 
-      String orderByTemplate = "${sqlWithoutOrderBy} ORDER BY ${orderBy} ${orderByDirection}";
+      String orderByTemplate = "${sqlQuery} ORDER BY ${orderBy} ${orderByDirection}";
       Map<String, String> orderByParams =
           ImmutableMap.<String, String>builder()
-              .put("sqlWithoutOrderBy", sqlWithoutOrderBy)
+              .put("sqlQuery", sqlQuery)
               .put("orderBy", orderByExpression)
               .put("orderByDirection", query.orderByDirection().name())
               .build();
-      return StringSubstitutor.replace(orderByTemplate, orderByParams);
-    } else {
-      return sqlWithoutOrderBy;
+      sqlQuery = StringSubstitutor.replace(orderByTemplate, orderByParams);
     }
+
+    if (query.pageSize() != null) {
+      String limitTemplate = "${sqlQuery} LIMIT ${pageSize}";
```

Comment:
I think calling this `pageSize` here and throughout this PR is not quite clear. Page size of 50 means that we return however many results there are in chunks of 50. Limit of 50 means we will return at most 50 results. Eventually, it would be nice to have both paging and a limit, so users can request e.g. the first 1000 results in chunks of 100. So I think it would be clearer to change the terminology from `pageSize` -> `limit` in this PR.